add community connected endpoint

tanneberger opened the pull request.

tanneberger committed:
add community connected endpoint

MarcelCoding commented on src/args.rs:
```diff
@@ -44,6 +44,13 @@ pub(crate) struct Args {
+    env = "WEBSITE_CONTENT_API_ALICE_LOOKING_GLASS_URL",
+    default_value = "https://lg.dd-ix.net/"
+  )]
+  pub(crate) alice_looking_glas_url: Url,
```
```suggestion
  pub(crate) alice_looking_glass_url: Url,
```

MarcelCoding commented on src/looking_glass.rs:
```diff
@@ -0,0 +1,136 @@
+      .client
+      .get(format!(
+        "https://{}/api/v1/routeservers/rs01_v4/neighbors",
+        self.looking_glass_url
```
looking_glass_url is only a address and not an url

MarcelCoding commented on src/looking_glass.rs:
```diff
@@ -0,0 +1,136 @@
+  }
+}
+
+pub fn address_in_network(routes: &[IpNet], network: IpAddr) -> bool {
```
```suggestion
pub fn is_address_in_network(routes: &[IpNet], network: IpAddr) -> bool {
```

MarcelCoding commented on src/looking_glass.rs:
```diff
@@ -0,0 +1,136 @@
+}
+
+pub fn address_in_network(routes: &[IpNet], network: IpAddr) -> bool {
+  warn!("routes size: {}", routes.len());
```
Debug logging?

MarcelCoding commented on src/routes/looking_glass.rs:
```diff
@@ -0,0 +1,24 @@
+}
+
+pub(crate) async fn get_connected_to_community(
+  ConnectInfo(addr): ConnectInfo<SocketAddr>,
```
Hast du mal getestet, ob axum das von den X-Forwarded Headern hier direkt schon einträgt?

tanneberger replied:
hab ich nicht, aber ist glaube einfacher das erstmal zu deployen und zu schauen, welche IP-Adresse wirklich ankommt.

MarcelCoding replied:
Wegen mir, kannst aber auch mit einem curl gens lokal deployte testen

MarcelCoding commented on src/args.rs:
```diff
@@ -44,6 +44,13 @@ pub(crate) struct Args {
+  #[clap(
+    long,
+    env = "WEBSITE_CONTENT_API_ALICE_LOOKING_GLASS_URL",
+    default_value = "https://lg.dd-ix.net/"
```
This is not an address. Either fix the name or the value.

tanneberger replied:
@MarcelCoding naming suggestions?

MarcelCoding replied:
alice_looking_glass_address or alice_looking_glass_url

MarcelCoding committed:
misc: cleanup

MarcelCoding committed:
cargo: update deps

MarcelCoding committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/5ed4e25ab58fd4c028b59d5611e14ea64de51d23?narHash=sha256-TvODmeR2W7yX/JmOCmP%2BlAFNkTT7hAxYcF3Kz8SZV3w%3D' (2025-09-29)
  → 'github:NixOS/nixpkgs/3acb677ea67d4c6218f33de0db0955f116b7588c?narHash=sha256-A5LS0AJZ1yDPTa2fHxufZN%2B%2Bn8MCmtgrJDtxFxrH4S8%3D' (2025-11-13)
